### PR TITLE
Stories/lsa 2021

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: app/models/report_generators/lsa/fy2021/base.rb [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      else
+        @coc_code = @report.options['coc_code']
+      end
+      if @report.options['project_id'].delete_if(&:blank?).any?
+        @project_ids = @report.options['project_id'].delete_if(&:blank?).map(&:to_i)
+        # Limit to only those projects the user who queued the report can see
+        # and to only those that the LSA can handle
+        @project_ids &= GrdaWarehouse::Hud::Project.viewable_by(@report.user).
+          in_coc(coc_code: @coc_code).
+          with_hud_project_type([1, 2, 3, 8, 9, 10, 13]).
```
Comment (awisema) [inline app/models/report_generators/lsa/fy2021/base.rb:39]: Make `LSA_PROJECT_TYPES` a constant?

File: app/models/reports/lsa/fy2021/base.rb [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        GrdaWarehouse::Hud::ProjectCoc.joins(project: :organization).
+        includes(project: :funders).
+        distinct.
+        merge(GrdaWarehouse::Hud::Project.viewable_by(user).coc_funded.hud_residential).
+        where(ProjectID: GrdaWarehouse::Hud::Enrollment.open_during_range(@range).select(:ProjectID)). # this is imperfect, but only look at projects with enrollments open during the past three years
+        where(Geocode: nil, geocode_override: nil),
+      )
+    end
+
+    private def geography_types(user)
```
Comment (awisema) [inline app/models/reports/lsa/fy2021/base.rb:169]: Is there a reason that some of these start with `missing_` and some don't?

File: drivers/hmis_csv_twenty_twenty_two/app/models/hmis_csv_twenty_twenty_two/exporter/client.rb [hunk 3]
```diff
@@ -30,17 +30,29 @@ def export! client_scope:, path:, export:
...
+        :SSNDataQuality,
+        :DOBDataQuality,
+        :GenderNone,
+        :Female,
+        :Male,
+        :NoSingleGender,
+        :Transgender,
+        :Questioning,
+        :VeteranStatus,
+        :NativeHIPacific,
```
Comment (awisema) [inline drivers/hmis_csv_twenty_twenty_two/app/models/hmis_csv_twenty_twenty_two/exporter/client.rb:50]: The other Race columns aren't required?

**Review by awisema — APPROVED** [review]

File: drivers/hmis_csv_twenty_twenty_two/app/models/hmis_csv_twenty_twenty_two/exporter/client.rb (hunk 3 above)
Comment (eanders) [reply drivers/hmis_csv_twenty_twenty_two/app/models/hmis_csv_twenty_twenty_two/exporter/client.rb:50]: They are also required, but I think they were already.  This is a rename of an old field, so data from 2020 doesn't exist until we migrate.

File: app/models/reports/lsa/fy2021/base.rb (hunk 2 above)
Comment (eanders) [reply app/models/reports/lsa/fy2021/base.rb:169]: no reason that I know of.  This was a direct copy from 2020

**Pull request merged**